libutils/string_lib.c

```diff
@@ -1801,3 +1801,27 @@ char *StringJoin(const Seq *const seq, const char *sep)
     char *const data = StringWriterClose(writer);
     return data;
 }
+
+Seq *StringSplit(const char *const str, const char *const del)
+{
+    assert(str != NULL);
+    assert(del != NULL);
+
+    Seq *const seq = SeqNew(1, free);
+
+    const char *start = str;
+    const char *end = strpbrk(str, del);
+
+    while (end != NULL)
+    {
+        char *const tmp = xstrndup(start, end - start);
```

Comment:
Why `const` for a temp variable that you only use once, right below. The variable doesn't even need to exist. Guys, this is just a bit too much, don't you think???

Reply:
Because [CONTRIBUTING.md](https://github.com/cfengine/core/blob/master/CONTRIBUTING.md#safety--correctness). The `const` keyword expresses that this pointer will not be reassigned.

Reply:
Yes, I know what `const` does and means. But using it for every single f*cking variable even a one that is just a temporary variable only assigned and used once in such a small scope is ridiculous.